utils/syscall.c

```diff
@@ -382,9 +382,343 @@ static const myst_syscall_pair_t _pairs[] = {
+      SYS_rt_tgsigqueueinfo,
+      SYS_execveat}},
+    {"signal",
+     14,
+     {SYS_rt_sigaction,
+      SYS_rt_sigprocmask,
+      SYS_rt_sigreturn,
+      SYS_pause,
+      SYS_kill,
+      SYS_rt_sigpending,
+      //   SYS_t_sigtimedwait,
+      SYS_rt_sigqueueinfo,
+      SYS_rt_sigsuspend,
+      SYS_sigaltstack,
+      SYS_tkill,
+      SYS_tgkill,
+      SYS_signalfd,
+      SYS_signalfd4,
+      SYS_rt_tgsigqueueinfo}},
+    {"memory",
+     24,
+     {SYS_mmap,
+      SYS_mprotect,
+      SYS_munmap,
+      SYS_brk,
+      SYS_mremap,
+      SYS_msync,
+      SYS_mincore,
+      SYS_madvise,
+      SYS_shmat,
+      SYS_shmdt,
+      SYS_mlock,
+      SYS_munlock,
+      SYS_mlockall,
+      SYS_munlockall,
+      SYS_io_setup,
+      SYS_io_destroy,
+      SYS_remap_file_pages,
+      SYS_mbind,
+      SYS_set_mempolicy,
+      SYS_get_mempolicy,
+      SYS_migrate_pages,
+      SYS_move_pages,
+      SYS_mlock2,
+      SYS_pkey_mprotect}},
+    {"stat", 1, {SYS_stat}},
+    {"lstat", 1, {SYS_lstat}},
+    {"fstat", 3, {SYS_fstat, SYS_newfstatat, SYS_statx}},
+    {"stat_like",
+     5,
+     {SYS_stat, SYS_fstat, SYS_lstat, SYS_newfstatat, SYS_statx}},
+    {"statfs", 1, {SYS_statfs}},
+    {"fstatfs", 1, {SYS_fstatfs}},
+    {"statfs_like", 3, {SYS_ustat, SYS_statfs, SYS_fstatfs}},
+    {"pure",
+     8,
+     {SYS_getpid,
+      SYS_getuid,
+      SYS_getgid,
+      SYS_geteuid,
+      SYS_getegid,
+      SYS_getppid,
+      SYS_getpgrp,
+      SYS_gettid}},
+    {"syscall_never_fails",
+     12,
+     {SYS_getpid,
+      SYS_umask,
+      SYS_getuid,
+      SYS_getgid,
+      SYS_geteuid,
+      SYS_getegid,
+      SYS_getppid,
+      SYS_getpgrp,
+      SYS_setfsuid,
+      SYS_setfsgid,
+      SYS_personality,
+      SYS_gettid}},
+    {"max_args", 0, {}},
+    {"memory_mapping_change",
+     11,
+     {SYS_mmap,
+      SYS_mprotect,
+      SYS_munmap,
+      SYS_brk,
+      SYS_mremap,
+      SYS_shmat,
+      SYS_execve,
+      SYS_shmdt,
+      SYS_remap_file_pages,
+      SYS_execveat,
+      SYS_pkey_mprotect}},
+    {"stackcapture_on_enter",
+     4,
+     {SYS_execve, SYS_exit, SYS_exit_group, SYS_execveat}},
+    {"compat_syscall_types", 0, {}},
+    {"seccomp_default", 2, {SYS_execve, SYS_execveat}},
+    {"creds",
+     19,
+     {SYS_getuid,
+      SYS_getgid,
+      SYS_setuid,
+      SYS_setgid,
+      SYS_geteuid,
+      SYS_getegid,
+      SYS_setreuid,
+      SYS_setregid,
+      SYS_getgroups,
+      SYS_setgroups,
+      SYS_setresuid,
+      SYS_getresuid,
+      SYS_setresgid,
+      SYS_getresgid,
+      SYS_setfsuid,
+      SYS_setfsgid,
+      SYS_capget,
+      SYS_capset,
+      SYS_prctl}},
+    {"clock",
+     7,
+     {SYS_gettimeofday,
+      SYS_adjtimex,
+      SYS_settimeofday,
+      SYS_time,
+      SYS_clock_settime,
+      SYS_clock_gettime,
+      //   SYS_lock_getres,
+      SYS_clock_adjtime}},
+    {"comm_change", 3, {SYS_execve, SYS_prctl, SYS_execveat}},
+    {"", 0, {}},
```

Comment:
Should this be `{NULL, 0, {}}`?

Reply:
So I had a test for this scenario that passed (even without the NULL element) which is why I didn't catch it earlier. See test 8 in the tests section.

Reply:
It seems to work without needing the NULL element at the end, but since this is more correct I'll add it